azurerm/internal/services/containers/kubernetes_cluster_node_pool_resource.go

```diff
@@ -647,6 +664,8 @@ func resourceKubernetesClusterNodePoolRead(d *pluginsdk.ResourceData, meta inter
 		d.Set("enable_auto_scaling", props.EnableAutoScaling)
 		d.Set("enable_node_public_ip", props.EnableNodePublicIP)
 		d.Set("enable_host_encryption", props.EnableEncryptionAtHost)
+		d.Set("fips_enabled", props.EnableFIPS)
+		d.Set("kubelet_disk_type", string(props.KubeletDiskType))
```

Comment:
this won't be returned for clusters provisioned with earlier API's and so will be an empty string, which'll lead to this being ForceNew here, can we confirm this by provisioning a cluster using the current provider version then updating that using this branch?

Reply:
thanks for pointing it. I tested use old azurerm version, but it's actually not returned. 

I have removed the default value and make it O+C.